client/client.go

```diff
@@ -127,3 +127,30 @@ func (client *Client) SysInfo() (*SysInfo, error) {
 
 	return &sysInfo, nil
 }
+
+type Capability struct {
+	Name  string            `json:"name"`
+	Label string            `json:"label"`
+	Type  string            `json:"type"`
+	Attrs map[string]string `json:"attrs,omitempty"`
+}
+
+// GetCapabilities gets the list of capabilities from the REST API
+func (client *Client) GetCapabilities() ([]Capability, error) {
+	var rsp response
+	if err := client.do("GET", "/1.0/capabilities", nil, &rsp); err != nil {
+		return nil, err
+	}
+	if rsp.Type == "error" {
+		// TODO: handle structured errors
```

Comment:
What's the status quo here?  Do we have a structured error format already, and if so why are we not unpacking it already?  If not, we need to add that to the upcoming list of things to fix.